### ISSUE TITLE
ODS 13 Error - requested 8183 bytes, but only 8153 bytes available

### ISSUE THREAD

**Issue by gerdus** [NeoZX/fb_formats#1]
I am getting following error when trying to parse firebird 4.0 database.
Have tried it on 2 different databases that work fine otherwise so I don't think it is db corruption.
Thank you.
```
Traceback (most recent call last):
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\go.py", line 3, in <module>
    d = Ods13.from_file("C:\\smartit3_databases\\DEMO.FDB4")
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\Lib\site-packages\kaitaistruct.py", line 44, in from_file
    return cls(KaitaiStream(f))
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 99, in __init__
    self._read()
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 108, in _read
    self.pages.append(Ods13.Page(_io__raw_pages, self, self._root))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 181, in __init__
    self._read()
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 193, in _read
    self.data = Ods13.IndexBTreePage(self._io, self, self._root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 512, in __init__
    self._read()
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\ods13.py", line 525, in _read
    self.btree_nodes = self._io.read_bytes(self.length)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Dev\2021\zedfire\toolstemp\py0ds13\Lib\site-packages\kaitaistruct.py", line 326, in read_bytes
    raise EOFError(
EOFError: requested 8183 bytes, but only 8153 bytes available
```

**Comment by NeoZX** [comment]
Could you provide a copy of the DB or metadata?

**Comment by gerdus** [comment]
Link to database: https://infireal.com/github-issues/DEMO.zip

**Comment by NeoZX** [comment]
Fixed 93b683e

**Comment by gerdus** [comment]
thank you for the quick fix!
